### ISSUE TITLE
[Feature]: Add GPU support on MacBooks (MPS)

### ISSUE THREAD

**Issue by fritshermans** [dynamiq-ai/pragmatiq#21]
### Problem

Thanks a lot for this great repository. I really enjoy studying and using it.

When I run the code on my MacBook, `pragmatiq info` tells me that CPUs are used. However, on my MacBook I should be able to use GPUs with the MPS backend of PyTorch (https://docs.pytorch.org/docs/2.14/notes/mps.html). It would be nice if the package could take advantage of GPUs on macs as well.

Keep up the good work!

### Proposed solution

Running any cli command with `pragmatiq` should automatically use MPS as torch device type if available.

### Area

Training or finetuning
